egs/fisher_swbd/s5/local/nnet3/run_tdnn_discriminative.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+
+set -o pipefail
+set -e
+# this is run_discriminative.sh
+
+# This script does discriminative training on top of CE nnet3 system.
+# note: this relies on having a cluster that has plenty of CPUs as well as GPUs,
+# since the lattice generation runs in about real-time, so takes of the order of
+# 1000 hours of CPU time.
+# 
+. cmd.sh
+
+
+stage=0
+train_stage=-10 # can be used to start training in the middle.
+get_egs_stage=-10
+use_gpu=true  # for training
+cleanup=false  # run with --cleanup true --stage 6 to clean up (remove large things like denlats,
+               # alignments and degs).
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+srcdir=exp/nnet3/tdnn_sp
+train_data_dir=data/train_nodup_sp_hires
+online_ivector_dir=exp/nnet3/ivectors_train_nodup_sp
+degs_dir=                     # If provided, will skip the degs directory creation
+lats_dir=                     # If provided, will skip denlats creation
+
+## Objective options
+criterion=smbr
+one_silence_class=true
+
+dir=${srcdir}_${criterion}
+
+## Egs options
+frames_per_eg=150
+frames_overlap_per_eg=30
+truncate_deriv_weights=10
+
+## Nnet training options
+effective_learning_rate=0.0000125
+max_param_change=1
+num_jobs_nnet=4
+num_epochs=4
```

Comment:
I see. I'll try changing #epoch and lrate. Thanks!

在 2016/4/6 12:38, Vimal Manohar 写道:
>
> In egs/fisher_swbd/s5/local/nnet3/run_tdnn_discriminative.sh 
> <https://github.com/kaldi-asr/kaldi/pull/657#discussion_r58650270>:
>
> > +## Objective options
> > +criterion=smbr
> > +one_silence_class=true
> > +
> > +dir=${srcdir}_${criterion}
> > +
> > +## Egs options
> > +frames_per_eg=150
> > +frames_overlap_per_eg=30
> > +truncate_deriv_weights=10
> > +
> > +## Nnet training options
> > +effective_learning_rate=0.0000125
> > +max_param_change=1
> > +num_jobs_nnet=4
> > +num_epochs=4
>
> The num_epochs might itself be reduced since there is a lot of data 
> and performance degrades after 1 epoch itself.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/kaldi-asr/kaldi/pull/657/files/a78789a804b2c83402703127757216f5e598ca29#r58650270>
>